File: 2713-114-1769-minimum-number-of-operations-to-move-all-balls-to-each-box/2713-114-1769-minimum-number-of-operations-to-move-all-balls-to-each-box.java
import java.util.*;

class Solution {
    public int[] minOperations(String boxes) {
        int n = boxes.length();
        int op = 0, L = 0, R = 0;

        for (int i = 0; i < n; i++) {
            if (boxes.charAt(i) == '1') {
                op += i;
                R++;
            }
        }

        int[] result = new int[n];
        for (int i = 0; i < n; i++) {
            result[i] = op;
            if (boxes.charAt(i) == '1') {
                L++;
                R--;
            }
            op += -R + L;
        }

        return result;
    }
}

File: 2713-114-1769-minimum-number-of-operations-to-move-all-balls-to-each-box/2713-114-1769-minimum-number-of-operations-to-move-all-balls-to-each-box.js
var minOperations = function(boxes) {
    const n = boxes.length;
    let op = 0, L = 0, R = 0;

    for (let i = 0; i < n; i++) {
        if (boxes[i] === '1') {
            op += i;
            R++;
        }
    }

    const result = Array(n).fill(0);
    for (let i = 0; i < n; i++) {
        result[i] = op;
        if (boxes[i] === '1') {
            L++;
            R--;
        }
        op += -R + L;
    }

    return result;
};
